### DISCUSSION TITLE
[HELP] Se puede configurar un SLA en un grupo tipo channel?

### DISCUSSION THREAD

**Discussion by NicolasPLG** [Cotalker/documentation#206] [2. Questions and Answers] [answered]
Hola

Quisiera saber si puedo configurar slas en un grupo de tipo channel para que me envié mensajes de aviso en un intervalo de tiempo (15 dias o 7 dias), desde una fecha que ingresa el usuario.

Sé que puede ser con scheduler, pero si se reinicia el server, se reinicia el scheduler (hasta lo que sé).

Puedo usar un bot sleep, pero creo que aplicaría lo mismo que el sheduler.

O sólo SLA es para las SM, para replantear el proceso.

Gracias :)

**Comment by guillevalin** [comment, marked as answer]
@NicolasPLG la funcionalidad de SLAs es sólo válido dentro del contexto de una máquina de estado, o workflow.

Si quisieras realizar algo similar con channels, tu única opción es crear una rutina que funcione con un cron. Si se reinicia el server, la ejecución de los cron no se ve afectada (siempre se ejecutarán cada X minutos, horas o días, según la configuración).

**Reply by NicolasPLG** [reply]
hola @guillevalin, entiendo, pero no afectaría si por ejemplo: envio un survey hoy para que en 3 dias me envíe un mensaje, y entre esos 3 días se cae el server, por decri. No afectaría?

**Reply by guillevalin** [reply]
Es que tienes que desacoplar dos cosas muy distintas:
- Frecuencia de ejecución del schedule (cron)
- Regla de negocio que si es positiva, realiza una acción, y que si es negativa, no realiza una acción.

En español, podrías tener un cron que revise reglas de negocio cada 1 hora, y solo enviar alertas cuando las reglas de negocio son positivas (true), por ende bajo ese punto de vista da lo mismo que se caiga el servidor por ejemplo.

La regla de negocio que tu quieres es que si chequeo hoy, o en una hora más la fecha de creación del formulario versus la fecha actual será menor a 3 días, pero a los 3 días, con 1 horas, se ejecutará de nuevo el cron y ahí la fecha de creación será mayor a 3 días, por lo tanto se ejecutará la alerta, se entiende?

**Reply by NicolasPLG** [reply]
entiendo probaré lo que me dices y te contaré.

Cerraré esta discusión y si me surge de nuevo crearé otra y tomaré referencia esta

Gracias @guillevalin
